pkg/odo/cli/component/test.go

```diff
@@ -0,0 +1,96 @@
+package component
+
+import (
+	"fmt"
+	"path/filepath"
+	"reflect"
+
+	"github.com/pkg/errors"
+
+	devfileParser "github.com/openshift/odo/pkg/devfile/parser"
+	projectCmd "github.com/openshift/odo/pkg/odo/cli/project"
+	"github.com/openshift/odo/pkg/odo/genericclioptions"
+	odoutil "github.com/openshift/odo/pkg/odo/util"
+	"github.com/openshift/odo/pkg/odo/util/completion"
+	"github.com/spf13/cobra"
+	"k8s.io/kubectl/pkg/util/templates"
+)
+
+// RecommendedCommandName is the recommended command name
+const TestRecommendedCommandName = "test"
+
+// TestOptions encapsulates the options for the odo command
+type TestOptions struct {
+	commandName      string
+	componentContext string
+	devfilePath      string
+	// testCommand      common.DevfileCommand
+	show bool
+	*genericclioptions.Context
+}
+
+var testExample = templates.Examples(`
+  # Run default test command
+  %[1]s
+
+  # Run a specific test command
+  %[1]s --test-command <command name>
+
+`)
+
+// NewTestOptions creates a new TestOptions instance
+func NewTestOptions() *TestOptions {
+	return &TestOptions{}
+}
+
+// Complete completes TestOptions after they've been created
+func (to *TestOptions) Complete(name string, cmd *cobra.Command, args []string) (err error) {
+	to.devfilePath = filepath.Join(to.componentContext, DevfilePath)
+	to.Context = genericclioptions.NewDevfileContext(cmd)
+
+	return
+}
+
+// Validate validates the TestOptions based on completed values
+func (to *TestOptions) Validate() (err error) {
+	devObj, err := devfileParser.ParseAndValidate(to.devfilePath)
```

Comment:
Since we're parsing and validating here, is there a way to pass this `devObj` to `RunTestCommand()` rather than getting it again in `RunTestCommand()`

Reply:
updated